src/solidity-frontend/solidity_convert.cpp

```diff
@@ -0,0 +1,1898 @@
+    symbol.static_lifetime = false;
+    symbol.file_local = true;
+  }
+  symbol.is_extern = false;
+
+  // For state var decl, we look for "value".
+  // For local var decl, we look for "initialValue"
+  bool has_init =
+    ast_node.contains("value") || ast_node.contains("initialValue");
+  if(symbol.static_lifetime && !symbol.is_extern && !has_init)
+  {
+    symbol.value = gen_zero(t, true);
+    symbol.value.zero_initializer(true);
+  }
+
+  // 6. add symbol into the context
+  // just like clang-c-frontend, we have to add the symbol before converting the initial assignment
+  symbolt &added_symbol = *move_symbol_to_context(symbol);
+
+  // 7. populate init value if there is any
+  code_declt decl(symbol_expr(added_symbol));
+
+  if(has_init)
+  {
+    exprt val;
+    if(get_expr(ast_node["initialValue"], val))
+      return true;
+
+    solidity_gen_typecast(ns, val, t);
+
+    added_symbol.value = val;
+    decl.operands().push_back(val);
+  }
+
+  decl.location() = location_begin;
+
+  new_expr = decl;
+
+  return false;
+}
+
+bool solidity_convertert::get_function_definition(
+  const nlohmann::json &ast_node)
+{
+  // For Solidity rule function-definition:
+  // Order matters! do not change!
+  // 1. Check fd.isImplicit() --- skipped since it's not applicable to Solidity
+  // 2. Check fd.isDefined() and fd.isThisDeclarationADefinition()
+  if(
+    !ast_node
+      ["implemented"]) // TODO: for interface function, it's just a definition. Add something like "&& isInterface_JustDefinition()"
+    return false;
+
+  // Check intrinsic functions
+  if(check_intrinsic_function(ast_node))
+    return false;
+
+  // 3. Set current_scope_var_num, current_functionDecl and old_functionDecl
+  current_scope_var_num = 1;
+  const nlohmann::json *old_functionDecl = current_functionDecl;
+  current_functionDecl = &ast_node;
+  current_functionName = (*current_functionDecl)["name"].get<std::string>();
+
+  // 4. Return type
+  code_typet type;
+  if(get_type_description(ast_node["returnParameters"], type.return_type()))
+    return true;
+
+  // 5. Check fd.isVariadic(), fd.isInlined()
+  //  Skipped since Solidity does not support variadic (optional args) or inline function.
+  //  Actually "inline" doesn not make sense in Solidity
+
+  // 6. Populate "locationt location_begin"
+  locationt location_begin;
+  get_location_from_decl(ast_node, location_begin);
+
+  // 7. Populate "std::string id, name"
+  std::string name, id;
+  get_function_definition_name(ast_node, name, id);
+
+  // 8. populate "std::string debug_modulename"
+  std::string debug_modulename =
+    get_modulename_from_path(location_begin.file().as_string());
+
+  // 9. Populate "symbol.static_lifetime", "symbol.is_extern" and "symbol.file_local"
+  symbolt symbol;
+  get_default_symbol(symbol, debug_modulename, type, name, id, location_begin);
+
+  symbol.lvalue = true;
+  symbol.is_extern =
+    false; // TODO: hard coded for now, may need to change later
```

Comment:
Remember to open an issue for this.

Reply:
ACK. 
will do it when extending the new frontend to support more benchmarks.